web/package.json

```diff
@@ -6,8 +6,9 @@
     "start": "vite",
     "build": "tsc && vite build",
     "serve": "vite preview",
+    "generate": "openapi -i http://localhost:5001/openapi.json -o './src/api/generated' -c axios",
```

Comment:
Why `-c axios`? The `fetch` client is just as good, and doesn't require the `axios` package, right?

Reply:
Oh, did we already use axios?

Reply:
the axios stuff is much simpler to use IMO. Like returning json, and raising errors on <200 etc

Reply:
We use fetch in de4rm, and we get errors on:
```
400: `Bad Request`,
401: `Unauthorized`,
404: `Not Found`,
422: `Unprocessable Entity`,
500: `Internal Server Error`,
```
Looking at the diff/files in this PR, I think this is the same for axios.

---

Now, in "returning json" - I think you use the Services the same way:

```ts
const updatedSession = await SessionService.get(sessionId)
```

We don't have to unwrap it in any way, we just get the response model specified in FastAPI (or in the OpenAPI spec). 

---
So I'm not sure if you really need Axios here, I think it just adds a dependency. Worth looking into?

Reply:
Cool, perhaps the generated fetch code has been "improved" since we last used it.
Could be a different issue to replace axios with fetch then, and see what changes are needed.

Reply:
I think the differences were larger in the last OpenAPI TS client generator tool/library we used!